File: project/app/src/main/java/org/owntracks/android/injection/modules/ActivityModules/RegionsActivityModule.java
package org.owntracks.android.injection.modules.ActivityModules;

import android.support.v7.app.AppCompatActivity;

import org.owntracks.android.injection.modules.ActivityModule;
import org.owntracks.android.injection.qualifier.ActivityContext;
import org.owntracks.android.injection.scopes.PerActivity;
import org.owntracks.android.ui.map.MapMvvm;
import org.owntracks.android.ui.map.MapViewModel;
import org.owntracks.android.ui.region.RegionActivity;
import org.owntracks.android.ui.regions.RegionsActivity;
import org.owntracks.android.ui.regions.RegionsMvvm;
import org.owntracks.android.ui.regions.RegionsViewModel;

import dagger.Binds;
import dagger.Module;

@Module(includes = ActivityModule.class)
public abstract class RegionsActivityModule {

    @Binds
    @PerActivity
    abstract AppCompatActivity bindActivity(RegionsActivity a);

    @Binds abstract RegionsMvvm.ViewModel bindViewModel(RegionsViewModel viewModel);
}

File: project/app/src/main/java/org/owntracks/android/ui/regions/RegionsActivityModule.kt
package org.owntracks.android.ui.regions

import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

@InstallIn(ActivityComponent::class)
@Module
abstract class RegionsActivityModule {
//    @Binds
//    @ActivityScoped
//    abstract fun bindActivity(a: RegionsActivity?): AppCompatActivity?

    @Binds
    abstract fun bindViewModel(viewModel: RegionsViewModel): RegionsMvvm.ViewModel<RegionsMvvm.View>
}